core/pkg/service/flag-evaluation/flag_evaluator_test.go

```diff
@@ -96,7 +100,8 @@ func TestConnectService_ResolveAll(t *testing.T) {
 				t.Errorf("ConnectService.ResolveAll() error = %v, wantErr %v", err.Error(), tt.wantErr.Error())
 				return
 			}
-			data, err := exp.Collect(context.TODO())
+			var data metricdata.ResourceMetrics
```

Comment:
out of scope for this PR, but I believe we could improve here by creating an interface for the metrics recorder which can be mocked here. Right now we are also testing functionality of that component here, which exceeds the scope of the flag evaluation component in my opinion.

Reply:
Yes, we could introduce a mock for `MetricsRecorder`. But, given that we validate recorded data, mock will need functionality similar to the real implementation.

And regarding the change here - this is an api change to have out variable assignment instead of a return